packages/page/src/Trash/PageTrashItemHandler.php

```diff
@@ -0,0 +1,187 @@
+use Webmozart\Assert\Assert;
+
+/**
+ * @internal
+ */
+final class PageTrashItemHandler implements
+    StoreTrashItemHandlerInterface,
+    RestoreTrashItemHandlerInterface,
+    RestoreConfigurationProviderInterface
+{
+    /**
+     * @param iterable<PageMapperInterface> $pageMappers
+     */
+    public function __construct(
+        private TrashItemRepositoryInterface $trashItemRepository,
+        private PageRepositoryInterface $pageRepository,
+        private ContentNormalizerInterface $contentNormalizer,
+        private ContentMergerInterface $contentMerger,
+        private iterable $pageMappers,
+    )
+    {
+    }
+
+    public static function getResourceKey(): string
+    {
+        return PageInterface::RESOURCE_KEY;
+    }
+
+    public function store(object $resource, array $options = []): TrashItemInterface
+    {
+        Assert::isInstanceOf($resource, PageInterface::class);
+
+        $page = $resource;
+
+        $data = [
+            'parentUuid' => $page->getParent()?->getUuid(),
+            'webspaceKey' => $page->getWebspaceKey(),
+            'dimensionContents' => [],
+        ];
+
+        $restoreType = $options['locales'] ?? null ? 'translation' : null;
+
+        $titles = [];
+        /** @var array<string, PageDimensionContentInterface> $localizedDimensionContents */
+        $localizedDimensionContents = [];
+        /** @var PageDimensionContentInterface|null $unlocalizedDimensionContent */
+        $unlocalizedDimensionContent = null;
+        foreach ($page->getDimensionContents() as $dimensionContent) {
+            if (
+                DimensionContentInterface::CURRENT_VERSION !== $dimensionContent->getVersion()
+                && DimensionContentInterface::STAGE_LIVE !== $dimensionContent->getStage()
+            ) {
+                continue;
+            }
+
+            if (null === $dimensionContent->getLocale()) {
+                $unlocalizedDimensionContent = $dimensionContent;
+                continue;
+            }
+
+            $localizedDimensionContents[$dimensionContent->getLocale()] = $dimensionContent;
+        }
+
+        Assert::notNull($unlocalizedDimensionContent, 'Expected to find an unlocalized dimension content for the page.');
+        Assert::notEmpty($localizedDimensionContents, 'Expected to find at least one localized dimension content for the page.');
+        foreach ($localizedDimensionContents as $locale => $localizedDimensionContent) {
+            /** @var array<int, PageDimensionContent> $dimensionContents */
+            $dimensionContents = [$unlocalizedDimensionContent, $localizedDimensionContent];
+
+            $mergedDimensionContent = $this->contentMerger->merge(
+                new DimensionContentCollection(
+                    $dimensionContents,
+                    [
+                        'locale' => $locale,
+                        // TODO handle other stages?
+                        'stage' => DimensionContentInterface::STAGE_LIVE,
```

Comment:
```suggestion
                        'stage' => DimensionContentInterface::STAGE_DRAFT,
```

The current draft should be added to the trash. Live might not even exist.